test/optimizer/shared_hash_join.test

```diff
@@ -0,0 +1,74 @@
+statement ok
+CREATE TABLE person (i string)
+
+statement ok
+INSERT INTO person VALUES ('j'), ('d'), ('p');
+
+statement ok
+CREATE TABLE knows (i string, j string);
+
+statement ok
+INSERT INTO knows VALUES ('j','p'), ('p','d'), ('d','j');
+
+query III
+SELECT  p1.i, p2.i, p3.i
+FROM    person p1, person p2, person p3, knows k1, knows k2, knows k3
+WHERE   p1.i = k1.i
+    AND p2.i = k1.j
+    AND p2.i = k2.i
+    AND p3.i = k2.j
+    AND p3.i = k3.i
+    AND p1.i = k3.j
+----
+j	p	d
+d	j	p
+p	d	j
+
+require tpch
```

Comment:
If I'm not wrong, before running a tpch query you must load a tpch db with the desired scale factor
e.g.,
```
statement ok
CALL dbgen(sf=0.1)
```

LOGICAL_TRANSACTION will only show up for transactions, I'm not so sure why you would see one when trying to run this query